src/coreclr/System.Private.CoreLib/src/Internal/Runtime/InteropServices/InMemoryAssemblyLoader.cs

```diff
@@ -20,6 +24,9 @@ public static class InMemoryAssemblyLoader
         public static unsafe void LoadInMemoryAssembly(IntPtr moduleHandle, IntPtr assemblyPath)
         {
 #if TARGET_WINDOWS
+            if (!IsSupported)
+                return;
```

Comment:
This should throw - returning and doing nothing will very likely confuse the caller and cause bad things to happen.

Reply:
Is this actually a public method? I don't see it in any `ref` sources.

Reply:
It's not public in the sense that anybody can call it. Just like many other public APIs in CoreLib. This one is different from the many others that it has no managed caller - but there are native callers.

Reply:
Yes, I'll double check and describing the expected feature switch behavior to give the complete picture

1. The library will have the 'dangerous' API to be used by developers (in this case, we explicitly root this API so that it will be there and also suppress the warning so that the library build doesn't generate the warning for this API)
2. When a developer trims an application, we use the msbuild property to define what would happen to the API. We plan to set the default property of the msbuild property to unroot the API (since this API is meant to be called from unmanaged callers only, there will be no reference from managed code to this API). The 'unroot' behavior is defined when `IsSupported` is set to false (it will throw now as suggested by FB above). I will check with a sample to check how the throw looks like from the native side. The default behavior will also mean that developers will not see any trimmer warnings related to this API.
3. A developer has the ability to set the msbuild property to keep the API as is (`IsSupported ` will be true) after trimming. We assume the developer knows how to make the trimmed application work in this case (specifically, the managed API they are going to call from native side will exist). The trimmer warnings will now light up on this scenario to give some indication of the potential problem to the developer.

Reply:
> the managed API they are going to call from native side

My query/concern is more around the fact that for C++/CLI (and COM), users aren't the ones making the API call. We provide `ijwhost` that calls the managed API, the SDK and project system link in the host, and it is all (ideally) opaque to the user. As a developer, if I get some exception or trimmer warning about how a method might be missing around some API I'm not trying to use (especially a one that is just in SPCL and not a public framework reference), what does that mean to me?